.github/workflows/main.yml

```diff
@@ -0,0 +1,183 @@
+name: build
+
+on:
+  push:
+    paths-ignore:
+      - 'Vagrantfile'
```

Comment:
there are other files like markdown files we also want to ignore. would it be easier if we white list the docker related files that trigger the build instead?